packages/calcite-components/src/components/block/block.tsx

```diff
@@ -207,8 +238,22 @@ export class Block
   //
   // --------------------------------------------------------------------------
 
+  /** Fires when the component is requested to be closed and before the closing transition begins. */
+  @Event({ cancelable: false }) calciteBlockBeforeClose: EventEmitter<void>;
+
+  /** Fires when the component is added to the DOM but not rendered, and before the opening transition begins. */
+  @Event({ cancelable: false }) calciteBlockBeforeOpen: EventEmitter<void>;
+
+  /** Fires when the component is closed and animation is complete. */
+  @Event({ cancelable: false }) calciteBlockClose: EventEmitter<void>;
+
+  /** Fires when the component is open and animation is complete. */
+  @Event({ cancelable: false }) calciteBlockOpen: EventEmitter<void>;
+
   /**
    * Emits when the component's header is clicked.
+   *
+   * @deprecated use `openClose` events instead.
```

Comment:
Can we explicitly list the events? I'm not sure developers would know what `openClose` events are.